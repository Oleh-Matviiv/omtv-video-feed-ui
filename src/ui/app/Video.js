import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Container = styled.article.attrs({
  className: 'video',
})`
  border: .1rem solid #efefef;
  margin-bottom: 2rem;
  display: grid;
  grid-template-areas: "title views"
                       "play play";
`;
export const Title = styled.div.attrs({
  className: 'video-title',
})`
  padding: 1rem;
  grid-area: "title";
`;
export const Views = styled.div.attrs({
  className: 'video-views',
})`
  padding: 1rem;
  grid-area: "views";
  text-align: right;
`;
export const Play = styled.div.attrs({
  className: 'video-play',
})`
  grid-area: "play";
`;

const Video = ({
  title = '',
  type = 'video',
  source,
  video = 'video unavailable',
  views = 'no info',
}) => (
  <Container>
    <Title>{ title }</Title>
    <Views>{ views }</Views>
    <Play>{ video }</Play>
  </Container>
);

Video.propTypes = {
  title: PropTypes.string,
  type: PropTypes.oneOf(['video']),
  source: PropTypes.oneOf([
    'url', 'youtube', 'facebook',
  ]).isRequired,
  video: PropTypes.string,
  views: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

export default Video;
