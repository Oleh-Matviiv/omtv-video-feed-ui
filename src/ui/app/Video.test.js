import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';

import Video, { Container, Title, Views, Play } from './Video';

describe('snapshots', () => {
  it('Video url', () => {
    const wrapper = shallow(
      <Video
        title="url based video"
        source="url"
        video="http://g.com/1.mp4"
        views={3445}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Video youtube', () => {
    const wrapper = shallow(
      <Video
        title="youtube based video"
        source="youtube"
        video="Y34fdvsf43"
        views={214526}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Video facebook', () => {
    const wrapper = shallow(
      <Video
        title="facebook based video"
        source="facebook"
        video="F34fdvsf43"
        views={56789}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Container', () => {
    const wrapper = mount(
      <Container>children</Container>
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Title', () => {
    const wrapper = mount(
      <Title>title</Title>
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Views', () => {
    const wrapper = mount(
      <Views>24341</Views>
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Play', () => {
    const wrapper = mount(
      <Play>children</Play>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
