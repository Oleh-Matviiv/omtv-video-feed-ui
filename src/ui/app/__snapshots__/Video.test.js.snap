// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Container 1`] = `
.c0 {
  border: .1rem solid #efefef;
  margin-bottom: 2rem;
  display: grid;
  grid-template-areas: "title views" "play play";
}

<styled.article>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "video",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  border: .1rem solid #efefef;
  margin-bottom: 2rem;
  display: grid;
  grid-template-areas: \\"title views\\"
                       \\"play play\\";
",
          ],
        },
        "displayName": "styled.article",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "article",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <article
      className="video c0"
    >
      children
    </article>
  </StyledComponent>
</styled.article>
`;

exports[`snapshots Play 1`] = `
.c0 {
  grid-area: "play";
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "video-play",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  grid-area: \\"play\\";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="video-play c0"
    >
      children
    </div>
  </StyledComponent>
</styled.div>
`;

exports[`snapshots Title 1`] = `
.c0 {
  padding: 1rem;
  grid-area: "title";
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "video-title",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  padding: 1rem;
  grid-area: \\"title\\";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="video-title c0"
    >
      title
    </div>
  </StyledComponent>
</styled.div>
`;

exports[`snapshots Video facebook 1`] = `
<styled.article>
  <styled.div>
    facebook based video
  </styled.div>
  <styled.div>
    56789
  </styled.div>
  <styled.div>
    F34fdvsf43
  </styled.div>
</styled.article>
`;

exports[`snapshots Video url 1`] = `
<styled.article>
  <styled.div>
    url based video
  </styled.div>
  <styled.div>
    3445
  </styled.div>
  <styled.div>
    http://g.com/1.mp4
  </styled.div>
</styled.article>
`;

exports[`snapshots Video youtube 1`] = `
<styled.article>
  <styled.div>
    youtube based video
  </styled.div>
  <styled.div>
    214526
  </styled.div>
  <styled.div>
    Y34fdvsf43
  </styled.div>
</styled.article>
`;

exports[`snapshots Views 1`] = `
.c0 {
  padding: 1rem;
  grid-area: "views";
  text-align: right;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "video-views",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  padding: 1rem;
  grid-area: \\"views\\";
  text-align: right;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="video-views c0"
    >
      24341
    </div>
  </StyledComponent>
</styled.div>
`;
